src/index.js

```diff
@@ -1,34 +1,47 @@
 'use strict'
 
-const { serializeError } = require('serialize-error')
-const debug = require('debug-logfmt')('splashy')
-const createVibrant = require('./vibrant')
-
-const toPalette = swatch =>
-  Object.keys(swatch)
-    .reduce((acc, key) => {
-      const value = swatch[key]
-      if (value) {
-        acc.push({
-          popularity: value.getPopulation(),
-          hex: value.getHex()
-        })
-      }
-      return acc
-    }, [])
-    .sort((a, b) => a.popularity <= b.popularity)
-    .map(color => color.hex)
-
-module.exports = async input => {
-  let swatch
-
-  try {
-    const vibrant = createVibrant(input)
-    swatch = await vibrant.getPalette()
-  } catch (error) {
-    debug.error(serializeError(error))
-    swatch = {}
+const quantize = require('@lokesh.dhakar/quantize')
+const ndarray = require('ndarray')
+const sharp = require('sharp')
+
+async function getPixels (buffer) {
+  const { data, info } = await sharp(buffer)
+    .ensureAlpha()
+    .raw()
+    .toBuffer({ resolveWithObject: true })
+
+  return ndarray(
+    new Uint8Array(data.buffer, data.byteOffset, data.length),
+    [info.width, info.height, 4],
+    [4, (4 * info.width) | 0, 1],
+    0
+  )
+}
+
+function createPixelArray (pixels, pixelCount, quality = 10) {
+  const pixelArray = []
+
+  for (let i = 0, offset; i < pixelCount; i += quality) {
+    offset = i * 4
+    const r = pixels[offset]
+    const g = pixels[offset + 1]
+    const b = pixels[offset + 2]
+    const a = pixels[offset + 3]
+
+    if ((a === undefined || a >= 125) && !(r > 250 && g > 250 && b > 250)) {
```

Comment:
I think this code excludes black and white pixels. Any reason why we're doing that?


![image](https://github.com/user-attachments/assets/080bac49-5684-4b1c-94c7-b796edad6ae3)
![image](https://github.com/user-attachments/assets/b146c55c-a510-48ac-9ba2-80c0b6863085)

Reply:
This is interesting; it's in the original colorthief implementation
https://github.com/lokesh/color-thief/blob/master/src/color-thief-node.js#L17